### DISCUSSION TITLE
New AppIndicators spec

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2295] [General]
I'm opening this discussion in a hurry, being busy with other codes. Sorry for not having time for an investigation!

An email was kindly sent to me about https://gitlab.freedesktop.org/xdg/xdg-specs/-/issues/84

@palinek, @luis-pereira, would you please check whether we need to make changes to Status Notifier?

**Comment by darkshram** [comment]
GNOME Shell is having several issues with status notify icons [they removed support for them a few years back], which is a very well known issue affecting a lot of popular non-GNOME applications (Telegram Desktop, Discord, MS Teams, etc.). Seems, despite [the efforts](https://trac.transmissionbt.com/ticket/3685) of GNOME developers to kill the status notification icons, there are more than ever applications using status notification icons, specially electron-based applications. This means that hundreds of popular applications do not work properly in GNOME Shell. The old specs were 'declared' obsolete [by GNOME developers] because they are X11 specific. So, Freedesktop developers are proposing a new status notification spec [which seems will work both in X11 and Wayland], so these hundreds of applications may properly work within GNOME Shell.

They need the approval of GNOME, KDE and Ubuntu and suggested to include a few more parties. Linux Mint developers (Xapp) was suggested, but they already have their own spec in Xapp (XAppStatusIcon as an alternative to Gtk.StatusIcon and AppIndicator) which conflicts with the proposed spec.

So basically, they are asking to support and implement the new proposed spec and avoid becoming what [this cartoon exposes](https://xkcd.com/927/).

**Reply by tsujan** [reply]
That was one of my guesses too. But LXQt devs need to investigate it.

**Comment by redtide** [comment]
Is there a reason for this in particular? Just curious if there are new specifications other than StatusNotifierItem and AppIndicator, but looking at that issue seems there is none yet ATM.

I see that LXQt has already a StatusNotifierItem implementation in lxqt-qtplugin and lxqt-panel and KDE KStatusNotifierItem is still used.

**Reply by tsujan** [reply]
> Is there a reason for this in particular?

No, It seems to me that GNOME wants to make the problems that it has created for itself be others' problems.

**Reply by redtide** [reply]
Or are they working on [something new](https://gitlab.freedesktop.org/xdg/xdg-specs/-/merge_requests/54) that needs some time? (from the last post on the issue you linked above)

**Reply by tsujan** [reply]
Personally, I don't care. The DBus protocol is used everywhere outside GNOME, under Wayland too.

I opened this page only because a strange email was sent to me in that time, but it was about LXQt. Why it was sent to me, I don't have any idea.
